flake-modules/github-releases.nix

```
@@ -0,0 +1,96 @@
+{ config, lib, withSystem, ... }:
+{
+  options =
+    let
+      inherit (lib) mkOption mkOptionType types;
+    in
+    {
+      hercules-ci.github-releases = {
+        condition = lib.mkOption {
+          type = types.functionTo types.bool;
+          description = ''
+            Condition under which a release is going to be pushed.
+            This is a function accepting [HerculesCI parameters](https://docs.hercules-ci.com/hercules-ci-agent/evaluation#params-herculesCI)
+            and returning boolean.
+            By default, pushing happens if a tag is present.
+          '';
+          default = { tag, ... }: tag != null;
+          defaultText = ''
+            { tag, ... }: tag != null
+          '';
+        };
+        releaseTag = lib.mkOption {
+          type = types.functionTo types.str;
+          description = ''
+            Tag to be assigned to the release.
+          '';
+          default = herculesCI: herculesCI.config.repo.tag;
+          defaultText = "herculesCI: herculesCI.config.repo.tag";
+        };
+        files = lib.mkOption {
+          type = types.listOf types.str;
+          description = ''
+            List of asset _files_ --- no directories allowed.
+            Each path is in a form of `(/path/to/file|glob)[#display_label]`.
+          '';
+        };
+        pushJobName = lib.mkOption {
+          type = types.str;
+          description = ''
+            Name of the Hercules CI job in which to perform the deployment.
+            By default the GitHub pages deployment is triggered by the `onPush.default` job,
+            so that the deployment only proceeds when the default builds are successful.
+          '';
+          default = "default";
+          defaultText = "default";
+        };
+      };
+    };
+
+  config =
+    let
+      inherit (lib) mkIf mkMerge;
+      inherit (config) defaultEffectSystem;
+
+      cfg = config.hercules-ci.github-releases;
+      enable = cfg ? files;
```

```suggestion
      enable = cfg.files != [ ];
```

The option always exists.